gocourse05/security/security.go

```diff
@@ -0,0 +1,111 @@
+package security
+
+type Range float32
+type Angle float32
+
+type Rotater interface {
+	Rotate(angle Angle)
+}
+
+type Recorder interface {
+	Record(c Conditions, records *[]AnimalRecord, rec AnimalRecord)
+}
+
+type AnimalRecord struct {
+	AnimalType
+	AnimalAction
+}
+
+func NewAnimalRecord(t AnimalType, a AnimalAction) *AnimalRecord {
+	return &AnimalRecord{
+		AnimalType:   t,
+		AnimalAction: a,
+	}
+}
+
+type AnimalType string
+type AnimalAction string
+
+const (
+	Bear  AnimalType = "Bear"
+	Tiger AnimalType = "Tiger"
+	Rat   AnimalType = "Rat"
+
+	Sleep   AnimalAction = "Sleep"
+	Eat     AnimalAction = "Eat"
+	Walking AnimalAction = "Walking"
+)
+
+type Conditions struct {
```

Comment:
Краще без суфіксу `s`:
```suggestion
type Condition struct {
```

Зазвичай із суфіксом `s` пишуть колекцію типів.